### PR TITLE
added review a PR guide

### PR DESCRIPTION
**Pull request by shuyama1**
fixes b/288626199

Screenshot 2023-07-05 at 11 13 55 AM

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: docs/content/contribute/review-pr.md
Comment (rileykarson) [inline docs/content/contribute/review-pr.md:10]: If an issue isn't linked, we should have folks check the issue tracker and add resolved issues if any.

I'm not sure how to write it clearly in this guide, but ideally we'd either have every single PR either:

1. Link to an issue with context
2. Establish clear context itself. For simple things (minor fixups, minor test fixes) this may be the title, but in other cases we'd want details in the description. This would serve to tackle contextless PRs.

File: docs/content/contribute/review-pr.md
Comment (rileykarson) [inline docs/content/contribute/review-pr.md:6]: There are different expectations for maintainer and contributor PRs (self-service / community), should we call them out separately in some way? In particular I'm thinking of:

- I don't give more than a passing glance to release notes for a maintainer PR, and trust they've written them correctly. For contributor PRs I've got to author them myself.
- For contributors I've got to wait until all tests are green until clicking approve. For maintainers, I'm generally fine approving if I'm satisfied with the code as-written and tests are most of the way (80-90%) there, even if they're currently failing.

File: docs/content/contribute/review-pr.md
Comment (rileykarson) [inline docs/content/contribute/review-pr.md:25]: Warn that VCR doesn't run in GA?

File: docs/content/contribute/review-pr.md
Comment (rileykarson) [inline docs/content/contribute/review-pr.md:19]: I would be inclined to put this lower down- I've seen a few cases where folks have trusted tests because they were green, but that tested the wrong thing. I'd encourage something along the lines of "Verify the tests ensure that the linked issue was resolved, if any".

For example- a bugfix should test the bug (or explain why it's not feasible to do so in the description) and an enhancement should test the new behaviour(s).

File: docs/content/contribute/review-pr.md
Comment (rileykarson) [inline docs/content/contribute/review-pr.md:24]:
```suggestion
   1. all mutable fields are tested in at least one update test.
```

Comment (rileykarson) [reply docs/content/contribute/review-pr.md:24]: Also, maybe encourage updating from a non-zero value to a zero value?

File: docs/content/contribute/review-pr.md
Comment (rileykarson) [inline docs/content/contribute/review-pr.md:30]:
```suggestion
1. Check if release notes capture all changes in the pull request, and are correctly formatted following the guidance in [write release notes]({{< ref "release-notes" >}}).
```

**Comment by shuyama1** [comment]
updated version:
Screenshot 2023-07-09 at 11 46 45 PM

**Comment by rileykarson** [comment]
Any thoughts on different responsibilities as mentioned in https://github.com/GoogleCloudPlatform/magic-modules/pull/8258#discussion_r1253483189?

File: docs/content/contribute/review-pr.md
Comment (melinath) [reply docs/content/contribute/review-pr.md:6]: FWIW I would prefer not to call out that we might have separate standards for maintainers since that is not relevant to most of the people who would read this page.

Comment (rileykarson) [reply docs/content/contribute/review-pr.md:6]: Ehh, fair enough- the page is doing double-duty as a maintainer and contributor guide in that case, which may leave ambiguities for maintainers.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**
